src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -2069,6 +2069,134 @@ public IFileSystem FileSystem
 
         #endregion ResultsCachingTestsAndHelpers
 
+        [Fact]
+        public void CheckIncompatibleRules_DisableIncompatibleRuleAndContinueAnalysis()
+        {
+            IncompatibleTestRule[] skimmers = new[]
+            {
+                new IncompatibleTestRule("TEST1001", null),
+                new IncompatibleTestRule("TEST1002", null),
+                new IncompatibleTestRule("TEST1003", new HashSet<string> { "TEST1001" },  IncompatibleRuleHandling.DisableAndContinueAnalysis),
+            };
```

Comment:
suggestion:

if rule 1 is incompatible with rule 2,
the incompatible flag can be set at rule 1, or rule 2.
And they should be equivalent, is it correct.

What if they are both set, may add a test for that.
And then a test for rule 1 -> rule 2 -> rule 3 -> rule 1, circular case.